packages/preview-middleware-client/src/adp/controllers/AddFragment.controller.ts

```diff
@@ -296,8 +296,15 @@ export default class AddFragment extends BaseDialog<AddFragmentModel> {
 
     private getFragmentTemplateName(targetAggregation: string): string {
         const currentControlName = this.runtimeControl.getMetadata().getName();
-        return currentControlName === 'sap.uxap.ObjectPageLayout' && targetAggregation === 'sections'
-            ? 'OBJECT_PAGE_CUSTOM_SECTION'
-            : '';
+        if (currentControlName === 'sap.uxap.ObjectPageLayout' && targetAggregation === 'sections') {
+            return 'OBJECT_PAGE_CUSTOM_SECTION';
+        } else if (
+            (currentControlName === 'sap.m.FlexBox' && targetAggregation === 'items') ||
```

Comment:
This is a too generic condition. FlexBox is a common container and is not restricted to header, so this would match basically any FlexBox in the application. The check should be similar to the one in the Quick Action.

Reply:
I adapted the the condition, but as I don't have running application. not sure if this will work.